plugins/optimization-detective/class-od-url-metric-group.php

```diff
@@ -320,6 +320,62 @@ public function get_lcp_element(): ?OD_Element {
 		return $result;
 	}
 
+	/**
+	 * Gets all elements from all URL Metrics in the viewport group keyed by the elements' XPaths.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array<string, non-empty-array<int, OD_Element>> Keys are XPaths and values are the element instances.
+	 */
+	public function get_xpath_elements_map(): array {
+		$result = ( function () {
+			$all_elements = array();
+			foreach ( $this->url_metrics as $url_metric ) {
+				foreach ( $url_metric->get_elements() as $element ) {
+					$all_elements[ $element->get_xpath() ][] = $element;
+				}
+			}
+			return $all_elements;
+		} )();
+
+		return $result;
+	}
+
+	/**
+	 * Gets the max intersection ratios of all elements in the viewport group and its captured URL Metrics.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array<string, float> Keys are XPaths and values are the intersection ratios.
+	 */
+	public function get_all_element_max_intersection_ratios(): array {
```

Comment:
There is an existing `get_all_element_max_intersection_ratios` on the `OD_URL_Metric_Group_Collection` class. It could be refactored to instead loop over all of `groups` to then call this method and merge the results.

Reply:
Made the changes in aa06707. Also refactored the `get_xpath_elements_map` method in a similar fashion.